src/context/wallet/useWallet.tsx

```diff
@@ -28,27 +30,40 @@ export function useWallet() {
   useEffect(() => {
     // add this check to support older wallet versions
     if (isUpdatedVersion()) {
-      window.aptos.on("accountChanged", (account: any) => {
+      window.aptos?.on?.("accountChanged", (account: any) => {
```

Comment:
If window.aptos is not the object we expect it to be, should any of the code in this file be running?